cache/class-vary-cache.php

```diff
@@ -61,6 +61,7 @@ public static function remove_no_cache_for_user() {
 	 * @param  array $groups  One or more groups to vary on.
 	 */
 	public static function register_groups( $groups ) {
+		self::$groups = [];
```

Comment:
This is probably going to create problems in cases where different parts of a codebase (e.g. different plugins) need to register groups.

Reply:
Should be good to merge after this.

Reply:
Oops, that was left that in unintentionally when adding the group clear method